test_bot/cogs/misc.py

```diff
@@ -29,15 +31,17 @@ async def attachment_desc(self, inter: disnake.AppCmdInter, desc: str = "test")
         await inter.response.send_message(file=self._get_file(desc))
 
     @commands.slash_command()
-    async def attachment_desc_edit(self, inter: disnake.AppCmdInter, desc: str = "test") -> None:
+    async def attachment_desc_edit(
+        self, inter: disnake.AppCmdInter[commands.Bot], desc: str = "test"
+    ) -> None:
         """Send a message with a button, which sends an attachment with the given description (or the default)
 
         Parameters
         ----------
         desc: The attachment description
         """
         button = disnake.ui.Button(label="edit")
-        button.callback = lambda interaction: interaction.response.edit_message(
+        button.callback = lambda interaction: interaction.response.edit_message(  # type: ignore
```

Comment:
this type ignore could be avoided if we pass a default type in Item.callback ~~or if TypeVar(s) with a default were already a thing~~

Reply:
Unless I'm missing something, this can be resolved like `commands.Param`, using `MessageInteraction[ClientT]` in the `Item.callback` definition. `Modal.callback` is similar as well.

Honestly, at this point I'm no longer entirely opposed to using TypeVar defaults, even if the PEP is still a draft and was (unfortunately) postponed to 3.13. This is something for a future PR though, for now this should only use standardized typing features :>